### ISSUE TITLE
Error installing on Mac

### ISSUE THREAD

**Issue by qrpike** [sheldonth/activetick-addon#4]
I am on Mac OSX version: 10.11.4
Node version: 5.3.0

```
Quintons-MacBook-Pro:feeds qrpike$ npm install activetick-addon --save
▌ ╢░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░╟
WARN engine activetick-addon@0.1.0: wanted: {"node":"0.12.7"} (current: {"node":"5.3.0","npm":"3.5.2"})

> activetick-addon@0.1.0 preinstall /Users/qrpike/stackdot/feeds/node_modules/.staging/activetick-addon-eb3d5472
> bash ./scripts/install.sh

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1895k  100 1895k    0     0  2045k      0 --:--:-- --:--:-- --:--:-- 2045k

> activetick-addon@0.1.0 install /Users/qrpike/stackdot/feeds/node_modules/activetick-addon
> node-gyp rebuild && bash ./scripts/postinstall.sh

  CXX(target) Release/obj.target/NodeActiveTickAddon/NodeActiveTickAddon.o
  CXX(target) Release/obj.target/NodeActiveTickAddon/NodeActiveTick.o
../NodeActiveTick.cpp:93:26: error: no matching function for call to 'New'
  Local<Object> buffer = node::Buffer::New(data_length);
                         ^~~~~~~~~~~~~~~~~
/Users/qrpike/.node-gyp/5.3.0/include/node/node_buffer.h:28:40: note: candidate function not viable: requires 2 arguments, but 1 was provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate, size_t length);
                                       ^
/Users/qrpike/.node-gyp/5.3.0/include/node/node_buffer.h:31:40: note: candidate function not viable: requires at least 2 arguments, but 1 was provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/Users/qrpike/.node-gyp/5.3.0/include/node/node_buffer.h:43:40: note: candidate function not viable: requires 3 arguments, but 1 was provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/Users/qrpike/.node-gyp/5.3.0/include/node/node_buffer.h:36:40: note: candidate function not viable: requires 5 arguments, but 1 was provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
../NodeActiveTick.cpp:349:10: warning: unused variable 'time' [-Wunused-variable]
  ATTIME time = pResponse->serverTime;
         ^
../NodeActiveTick.cpp:347:11: warning: unused variable 'perm' [-Wunused-variable]
  uint8_t perm = pResponse->permissions[0];
          ^
../NodeActiveTick.cpp:389:28: error: no matching function for call to 'New'
    Local<Object> buffer = node::Buffer::New(m->data_sz);
                           ^~~~~~~~~~~~~~~~~
/Users/qrpike/.node-gyp/5.3.0/include/node/node_buffer.h:28:40: note: candidate function not viable: requires 2 arguments, but 1 was provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate, size_t length);
                                       ^
/Users/qrpike/.node-gyp/5.3.0/include/node/node_buffer.h:31:40: note: candidate function not viable: requires at least 2 arguments, but 1 was provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/Users/qrpike/.node-gyp/5.3.0/include/node/node_buffer.h:43:40: note: candidate function not viable: requires 3 arguments, but 1 was provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/Users/qrpike/.node-gyp/5.3.0/include/node/node_buffer.h:36:40: note: candidate function not viable: requires 5 arguments, but 1 was provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
2 warnings and 2 errors generated.
make: *** [Release/obj.target/NodeActiveTickAddon/NodeActiveTick.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/qrpike/.nvm/versions/node/v5.3.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 15.4.0
gyp ERR! command "/Users/qrpike/.nvm/versions/node/v5.3.0/bin/node" "/Users/qrpike/.nvm/versions/node/v5.3.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/qrpike/stackdot/feeds/node_modules/activetick-addon
gyp ERR! node -v v5.3.0
gyp ERR! node-gyp -v v3.2.1
gyp ERR! not ok 
feeds@1.0.0 /Users/qrpike/stackdot/feeds
```

**Comment by sheldonth** [comment]
Unfortunately the c++ bindings only work with node version 0.12.7 at the moment.

I am almost there implementing Nan so that a single C++ interface works across all versions of V8, but not quite.

**Comment by qrpike** [comment]
Please let me know when it's all done! Thanks

**Comment by qrpike** [comment]
Any updates on this? Could really use this library. 

Thanks,

**Comment by sheldonth** [comment]
Yeah, I actually started looking at the work remaining here over the weekend. I'll see if I can get the Nan adapters finished this week.

As a reminder this library will work on node 0.12.7 at the moment, however there are a few memory leaks in the C++ that are going to give you trouble if you use it for large amounts of data.

**Comment by sheldonth** [comment]
@qrpike What version of node are you trying to use? 4.\* ?

**Comment by qrpike** [comment]
@sheldonth I'm trying to use the latest which is 6.*

I'm trying to get streaming tick data for many many symbols, which the http library doesn't seem to like very much. Hence why I'm trying to use the library instead.

Thanks,

**Comment by sheldonth** [comment]
Gotcha. Okay, let me look at what parts of Nan aren't compiling in 6.\* -- yes you need to use their c++ interface to swallow data on large scale. I'll get back to you shortly.

**Comment by sheldonth** [comment]
Hi @qrpike 

I've updated the Nan interface such that I confirmed it compiles on Mac OS X in Node.js 6.6.0.

If you go to the /example folder and create a `config.js` that has a valid username/password/apikey you can then run either `iced example.iced` or `node example.js` to see a sample stream of realtime data.

I've also published npm version 0.1.1 that includes the fix for Nan compatibility. 

If you confirm it compiles on the newer node.js versions, I'd like to close this ticket and use another one for tracking the memory leak issues.

**Comment by qrpike** [comment]
@sheldonth Seems to be working great now. Compiling and working on macOS 10.12, node 6.3.1

Cheers!
